usd_qtpy/render_util/dialog.py

```diff
@@ -0,0 +1,889 @@
+        Returns a camera and whether it should be destroyed afterwards.
+        -> UsdGeom.Camera, should_destroy()
+        """
+
+        box_text = self.cbox_camera.currentText()
+        data = self.cbox_camera.currentData()
+
+        width = self.spinbox_horresolution.value()
+        height = self.spinbox_verresolution.value()
+
+        aspect_ratio: float = float(width) / float(height)
+
+        if data is not None:
+            data = UsdGeom.Camera(data)
+            v_aperture = data.GetVerticalApertureAttr()
+            # ensure that camera can render desired aspect ratio.
+            if not v_aperture.IsValid():
+                v_aperture = 24
+            else:
+                v_aperture = v_aperture.Get(0)
+
+            data.CreateHorizontalApertureAttr(v_aperture * aspect_ratio)
+            data.CreateHorizontalApertureOffsetAttr(0)
+            data.CreateVerticalApertureAttr(v_aperture)
+            data.CreateVerticalApertureOffsetAttr(0)
+
+            # Camera is in scene, doesn't need to be deleted.
+            return data, False
+
+        elif "Framing" in box_text:
+            camera = framing_camera.create_framing_camera_in_stage(
+                self._stage,
+                name="Playblast_framingCam",
+                fit=self.spinbox_fit.value(),
+                width=width,
+                height=height
+                )
+            return camera, True
+        elif "Viewer" in box_text:
+            camera = playblast.camera_from_stageview(self._stage, self._stageview, "Playblast_viewerCam")
+            # ensure camera aspect ratio
+            # Aperture size (24) is based on the size of a full frame SLR camera sensor
+            # https://en.wikipedia.org/wiki/Image_sensor_format#Common_image_sensor_formats
+            aspect_ratio: float = width / float(height)
+            camera.CreateHorizontalApertureAttr(24 * aspect_ratio)
+            camera.CreateHorizontalApertureOffsetAttr(0)
+            camera.CreateVerticalApertureAttr(24)
+            camera.CreateVerticalApertureOffsetAttr(0)
+            return camera, True
+        
+        return None, False
+
+    def playblast_callback(self):
+        frames = self._construct_frames_argument()
+        camera, should_destroy = self._get_camera()
+
+        path = self.txt_filename.text()
+
+        print(frames,camera,path)
+        if not path:
+            return
+
+        playblast.render_playblast(self._stage,
+                                   path,
+                                   frames,
+                                   self.spinbox_horresolution.value(),
+                                   camera,
+                                   self.cbox_complexity.currentText(),
+                                   self.cbox_renderer.currentText(),
+                                   "sRGB",
+                                   self._gather_purposes(),
+                                   self
+                                   )
```

Comment:
This is definitely clearer with the arguments named.

Reply:
I'll seperate the getting of the arguments from their respective UI elements and store them,
and also name them, `argname=argvariable`